packages/agreement-platformstate-writer/src/utils.ts

```diff
@@ -92,3 +111,434 @@ export const deleteAgreementEntry = async (
   const command = new DeleteItemCommand(input);
   await dynamoDBClient.send(command);
 };
+
+export const updateAgreementStateInPlatformStatesEntry = async (
+  dynamoDBClient: DynamoDBClient,
+  primaryKey: PlatformStatesAgreementPK,
+  state: ItemState,
+  version: number
+): Promise<void> => {
+  const input: UpdateItemInput = {
+    ConditionExpression: "attribute_exists(PK)",
+    Key: {
+      PK: {
+        S: primaryKey,
+      },
+    },
+    ExpressionAttributeValues: {
+      ":newState": {
+        S: state,
+      },
+      ":newVersion": {
+        N: version.toString(),
+      },
+      ":newUpdateAt": {
+        S: new Date().toISOString(),
+      },
+    },
+    ExpressionAttributeNames: {
+      "#state": "state",
+    },
+    UpdateExpression:
+      "SET #state = :newState, version = :newVersion, updatedAt = :newUpdateAt",
+    TableName: config.tokenGenerationReadModelTableNamePlatform,
+    ReturnValues: "NONE",
+  };
+  const command = new UpdateItemCommand(input);
+  await dynamoDBClient.send(command);
+};
+
+export const agreementStateToItemState = (state: AgreementState): ItemState =>
+  state === agreementState.active ? itemState.active : itemState.inactive;
+
+export const updateAgreementStateInTokenGenerationStatesTableEntries = async ({
```

Comment:
```suggestion
export const updateAgreementOnTokenStates = async ({
```

Reply:
Renamed to `updateAgreementStateOnTokenStatesEntries`. I am following this pattern: the functions ending with `_Entries` take the entries that have to be updated, as a parameter. The functions not ending with `_Entries` do the retrieval inside the function.